### DISCUSSION TITLE
Can we use `int` instead of `frontend.Variable` in circuit schema

### DISCUSSION THREAD

**Discussion by shreyas-londhe** [Consensys/gnark#351] [Q&A] [answered]
I have a usecase where I need to index an array inside the circuit using a variable defined in the circuit schema. If I make the variable of type `frontend.Variable` its complaining that arrray indexing can only be done with `int`s. 

If I make the variable `int`, then it always evaluates to 0 inside the circuit so 0th element is always picked from the array. 

Can someone please suggest a workaround for this, or how to solve this issue?

Thanks!

**Comment by gbotrel** [comment, marked as answer]
Array indexing inside a circuit is not supported yet. If you have limited number of elements, you can play with the `Select` or `Lookup` apis -- .

**Reply by shreyas-londhe** [reply]
Thank you so much, I'll look into this
